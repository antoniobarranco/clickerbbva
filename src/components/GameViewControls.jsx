import React from 'react';
import PropTypes from 'prop-types';

function GameViewControls({
  hitHandle, autoClickerHandle, megaClickerHandle, autoClickers,
  megaClickers, canBuyAutoClicker, canBuyMegaClicker,
}) {
  return (
    <div className="d-grid gap-2 col-6  my-5 mx-auto">
      <button type="button" className="btn btn-primary btn-lg" onClick={hitHandle}>Hit</button>
      <button type="button" aria-disabled={!canBuyAutoClicker} disabled={!canBuyAutoClicker} className="btn btn-warning btn-lg" onClick={autoClickerHandle}>
        <i className="bi bi-star" />
        {' '}
        Buy AutoClicker
        {' '}
        <span className="badge text-bg-light">
          x
          {autoClickers}
        </span>
      </button>
      <button type="button" aria-disabled={!canBuyMegaClicker} disabled={!canBuyMegaClicker} className="btn btn-danger btn-lg" onClick={megaClickerHandle}>
        <i className="bi bi-star-fill" />
        {' '}
        Buy MegaClicker
        {' '}
        <span className="badge text-bg-light">
          x
          {megaClickers}
        </span>
      </button>
    </div>
  );
}

GameViewControls.propTypes = {
  hitHandle: PropTypes.func.isRequired,
  autoClickerHandle: PropTypes.func.isRequired,
  megaClickerHandle: PropTypes.func.isRequired,
  autoClickers: PropTypes.number.isRequired,
  megaClickers: PropTypes.number.isRequired,
  canBuyAutoClicker: PropTypes.bool.isRequired,
  canBuyMegaClicker: PropTypes.bool.isRequired,
};

export default GameViewControls;
