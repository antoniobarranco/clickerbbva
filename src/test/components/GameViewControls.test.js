import renderer from 'react-test-renderer';
import { render, fireEvent, screen } from "@testing-library/react";
import GameViewControls from '../../components/GameViewControls';

describe('Tests on GameViewControls component', () => {

    test('Test GameViewControls with Snapshot test for UI changes', () => {

        const hitHandle = jest.fn();
        const autoClickerHandle = jest.fn();
        const megaClickerHandle = jest.fn();
        const autoClickers = 14;
        const megaClickers = 500;
        const canBuyAutoClicker = true;
        const canBuyMegaClicker = true;

        const tree = renderer.create(
            <GameViewControls
                hitHandle={hitHandle}
                autoClickerHandle={autoClickerHandle}
                megaClickerHandle={megaClickerHandle}
                autoClickers={autoClickers}
                autoClickers={megaClickers}
                autoClickers={canBuyAutoClicker}
                autoClickers={canBuyMegaClicker}
            />).toJSON();

        expect(tree).toMatchSnapshot();

    })

    test('Test GameViewControls Hit handles works propertly', () => {

        const hitHandle = jest.fn();
        const autoClickerHandle = jest.fn();
        const megaClickerHandle = jest.fn();
        const autoClickers = 14;
        const megaClickers = 500;
        const canBuyAutoClicker = true;
        const canBuyMegaClicker = true;

        render(<GameViewControls
                hitHandle={hitHandle}
                autoClickerHandle={autoClickerHandle}
                megaClickerHandle={megaClickerHandle}
                autoClickers={autoClickers}
                autoClickers={megaClickers}
                autoClickers={canBuyAutoClicker}
                autoClickers={canBuyMegaClicker}
            />);

        fireEvent.click(screen.getByTestId('HitButton'));
        expect(hitHandle).toHaveBeenCalledTimes(1);

    })


})