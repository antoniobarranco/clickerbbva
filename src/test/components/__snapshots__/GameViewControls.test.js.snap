// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on GameViewControls component Test GameViewControls with Snapshot test for UI changes 1`] = `
<div
  className="d-grid gap-2 col-6  my-5 mx-auto"
>
  <button
    className="btn btn-primary btn-lg"
    data-testid="HitButton"
    onClick={[MockFunction]}
    type="button"
  >
    Hit
  </button>
  <button
    aria-disabled={true}
    className="btn btn-warning btn-lg"
    data-testid="autoClickerButton"
    disabled={true}
    onClick={[MockFunction]}
    type="button"
  >
    <i
      className="bi bi-star"
    />
     
    Buy AutoClicker
     
    <span
      className="badge text-bg-light"
    >
      x
    </span>
  </button>
  <button
    aria-disabled={true}
    className="btn btn-danger btn-lg"
    data-testid="megaClickerButton"
    disabled={true}
    onClick={[MockFunction]}
    type="button"
  >
    <i
      className="bi bi-star-fill"
    />
     
    Buy MegaClicker
     
    <span
      className="badge text-bg-light"
    >
      x
    </span>
  </button>
</div>
`;
